ykrt/src/trace/mod.rs

```diff
@@ -57,7 +57,13 @@ pub(crate) trait TraceRecorder {
     fn stop(self: Box<Self>) -> Result<Box<dyn AOTTraceIterator>, InvalidTraceError>;
 }
 
-/// An iterator which [TraceRecord]s use to process a trace into [TraceAction]s.
+/// An iterator which [TraceRecord]s use to process a trace into [TraceAction]s. The iterator must
+/// respect the following:
+///
+/// 1. The control point call must not appear at the head of the trace.
+/// 2. Consecutive `TraceAction`s must not compare equal (i.e. the iterator must have deduplicated
+///    consecutive `TraceAction`s).
+/// 3. The "stop tracing" call must not appear at the tail of the trace.
```

Comment:
More precisely: the unmappable block containing the body of the "stop tracing" function must not appear at the tail of the trace. It's OK (and expected) for the prior unmappable block containing the call-site to the "stop tracing" function to be at the tail of the trace. I think that's right?

Reply:
You tell me!

Reply:
I believe what I've said is correct. Did the old code strip any mappable blocks from the end? If not, I think that proves it.

Reply:
In fairness, the code in this PR a) does what you're suggesting b) documents it in detail at line 48. So I think we're good?